### PR TITLE
Add ILI9341 driver rotation, fix MicroGraphics rotation logic, fix Tf…

### PR DESCRIPTION
**Pull request by adrianstevens**
…tSpi API typo + minor cleanup

### CODE REVIEW

File: Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroGraphics/Driver/MicroGraphics.cs [hunk 1]
```diff
@@ -19,7 +19,15 @@ public partial class MicroGraphics
         /// <summary>
         /// PixelBuffer draw target
         /// </summary>
-        protected IPixelBuffer pixelBuffer;
+        protected IPixelBuffer PixelBuffer
+        {
+            get
+            {
+                if (display != null) return display.PixelBuffer;
```
Comment (jorgedevs) [inline Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroGraphics/Driver/MicroGraphics.cs:26]: By our coding conventions we always add { } even for one line ifs

File: Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroGraphics/Driver/MicroGraphics.cs [hunk 2]
```diff
@@ -34,10 +42,7 @@ public IFont CurrentFont
             get
             {
                 // lazy load
```
Comment (jorgedevs) [inline Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroGraphics/Driver/MicroGraphics.cs:44]: This comment can be removed.

File: Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroGraphics/Driver/MicroGraphics.cs [hunk 3]
```diff
@@ -19,7 +19,15 @@ public partial class MicroGraphics
...
+            get
+            {
+                if (display != null) return display.PixelBuffer;
+                return _memoryBuffer;
+            }
+        }
+        private readonly IPixelBuffer _memoryBuffer;
 
         /// <summary>
         /// Ingore pixels that are outside of the pixel buffer coordinate space
```
Comment (jorgedevs) [inline Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroGraphics/Driver/MicroGraphics.cs:33]: Ignore

File: Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroGraphics/Driver/MicroGraphics.cs [hunk 4]
```diff
@@ -87,14 +92,36 @@ public RotationType Rotation
...
+        /// Return the height of the display after accounting for the rotation
         /// </summary>
-        public int Height => Rotation == RotationType.Default || Rotation == RotationType._180Degrees ? pixelBuffer.Height : pixelBuffer.Width;
+        public int Height
+        {
+            get
+            {
+                if (display is IRotatableDisplay { } d)
+                {
+                    return display.Height;
```
Comment (jorgedevs) [inline Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroGraphics/Driver/MicroGraphics.cs:103]: Shouldnt this be `return d.Height;`?

File: Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroGraphics/Driver/MicroGraphics.cs [hunk 5]
```diff
@@ -87,14 +92,36 @@ public RotationType Rotation
...
+        /// Return the width of the display after accounting for the rotation
         /// </summary>
-        public int Width => Rotation == RotationType.Default || Rotation == RotationType._180Degrees ? pixelBuffer.Width : pixelBuffer.Height;
+        public int Width
+        {
+            get
+            {
+                if (display is IRotatableDisplay { } d)
+                {
+                    return display.Width;
```
Comment (jorgedevs) [inline Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroGraphics/Driver/MicroGraphics.cs:119]: Shouldnt this be return d.Width;?

File: Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroGraphics/Driver/MicroGraphics.cs (hunk 1 above)
Comment (jorgedevs) [reply Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroGraphics/Driver/MicroGraphics.cs:26]: Other option is do:

`return (display != null) ? display.PixelBuffer : _memoryBuffer; `

Comment (adrianstevens) [reply Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroGraphics/Driver/MicroGraphics.cs:26]: yeah I missed that - I'll fix

File: Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroGraphics/Driver/MicroGraphics.cs (hunk 3 above)
Comment (adrianstevens) [reply Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroGraphics/Driver/MicroGraphics.cs:33]: haha .. I'll do a global search for ingore :)

File: Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroGraphics/Driver/MicroGraphics.cs (hunk 4 above)
Comment (adrianstevens) [reply Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroGraphics/Driver/MicroGraphics.cs:103]: yeah can do ... or get rid of the d

**Review by jorgedevs — APPROVED** [review]

**Pull request merged**